dev/tests/mailchimp/tests/app/Ebizmarts/MailChimp/Model/ObserverTest.php

```diff
@@ -2,9 +2,58 @@
 
 class Ebizmarts_MailChimp_Model_ObserverTest extends PHPUnit_Framework_TestCase
 {
+    protected $_mailchimpObserverMock;
+    protected $_observerMock;
+    protected $_eventObserverMock;
+    protected $_creditMemoMock;
+    protected $_orderMock;
+    protected $_requestMock;
+    protected $_subscriberMock;
+    protected $_helperMock;
+
     public function setUp()
     {
         Mage::app('default');
+
+        $this->_mailchimpObserverMock = $this->getMockBuilder(Ebizmarts_MailChimp_Model_Observer::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getRequest', 'getSubscriberModel', 'makeHelper'))
+            ->getMock();
+
+        $this->_observerMock = $this->getMockBuilder(Varien_Event_Observer::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getEvent'))
+            ->getMock();
+
+        $this->_eventObserverMock = $this->getMockBuilder(Varien_Event::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getCreditmemo'))
+            ->getMock();
+
+        $this->_creditMemoMock = $this->getMockBuilder(Mage_Sales_Model_Order_Creditmemo::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getOrder', 'getAllItems'))
+            ->getMock();
+
+        $this->_orderMock = $this->getMockBuilder(Mage_Sales_Model_Order::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getCustomerEmail'))
+            ->getMock();
+
+        $this->_requestMock = $this->getMockBuilder(Mage_Core_Controller_Request_Http::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('getParam'))
+            ->getMock();
+
+        $this->_subscriberMock = $this->getMockBuilder(Mage_Newsletter_Model_Subscriber::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('loadByEmail'))
+            ->getMock();
+
+        $this->_helperMock = $this->getMockBuilder(Ebizmarts_MailChimp_Helper_Data::class)
+            ->disableOriginalConstructor()
+            ->setMethods(array('unsubscribeMember'))
+            ->getMock();
```

Comment:
Since this mocks are not used in all the tests of the class, would be best extracting the creation of them to functions and call them from the required tests.